astropy/modeling/functional_models.py

```diff
@@ -572,6 +572,92 @@ def fit_deriv(x, amplitude, x_0, fwhm):
         return [d_amplitude, d_x_0, d_fwhm]
 
 
+class Voigt1D(Fittable1DModel):
+    """
+    One dimensional model for the Voigt profile.
+
+    Parameters
+    ----------
+    pos : float
+        Position of the peak
+    amp_L : float
+        The Lorentzian amplitude
+    fwhm_L : float
+        The Lorentzian full width at half maximum
+    fwhm_G : float
+        The Gaussian full width at half maximum
+
+    See Also
+    --------
+    Gaussian1D, Lorentz1D
+
+    Notes
+    -----
+    Algorithm for the computation taken from
+    McLean, A. B., Mitchell, C. E. J. & Swanston, D. M. Implementation of an
+    efficient analytical approximation to the Voigt function for photoemission
+    lineshape analysis. Journal of Electron Spectroscopy and Related Phenomena
+    69, 125-132 (1994)
+    """
+
+    pos = Parameter()
+    amp_L = Parameter()
+    fwhm_L = Parameter()
+    fwhm_G = Parameter()
+
+    @staticmethod
+    def evaluate(x, pos, amp_L, fwhm_L, fwhm_G):
+        A = [-1.2150, -1.3509, -1.2150, -1.3509]
+        B = [1.2359, 0.3786, -1.2359, -0.3786]
+        C = [-0.3085, 0.5906, 0.3085, 0.5906]
+        D = [0.0210, -1.1858, -0.0210, 1.1858]
```

Comment:
These constants should be defined once as a class-level attribute.  I don't know what a great name would be so maybe just:
```python
_abcd = np.array([
    [-1.2150, -1.3509, -1.2150, -1.3509],  # A
    [1.2359, 0.3786, -1.2359, -0.3786],    # B
    [-0.3085, 0.5906, 0.3085, 0.5906],     # C
    [0.0210, -1.1858, -0.0210, 1.1858]])   # D